tests/php/ORM/ArrayLibTest.php

```diff
@@ -335,4 +335,17 @@ public function testIterateVolatileModified()
             'New items are iterated over'
         );
     }
+
+    public function testShuffleAssociative()
+    {
+        $list = ['a' => 1, 'b' => 2, 'c' => 3, 'd' => 4, 'e' => 5, 'f' => 6];
+        $copy = $list;
+        ArrayLib::shuffleAssociative($copy);
+        // Check value/key association is retained
+        foreach ($list as $key => $value) {
+            $this->assertEquals($value, $copy[$key]);
+        }
+        // Check the order is different
+        $this->assertNotSame($list, $copy);
```

Comment:
Is it possible that the shuffle randomly ends up with the same list as before? e.g shuffling an array [1,2] 50% of the time may end up back as [1,2] ?

Reply:
Possible but very unlikely. I don't think it's worth caring about but I can make it try a few times before failing if you want me to.